orchestrator/orchestrator.go

```diff
@@ -245,13 +251,24 @@ func (orch Orchestrator) EnqueueMissingEvents(
 func (orch Orchestrator) ProcessNonces(
 	ctx context.Context,
 	noncesQueue chan uint64,
+	requeueQueue chan uint64,
 	signalChan chan<- struct{},
 ) error {
+	ticker := time.NewTicker(time.Hour)
 	for {
 		select {
 		case <-ctx.Done():
 			close(signalChan)
 			return ErrSignalChanNotif
+		case <-ticker.C:
+			if len(requeueQueue) > 0 && len(noncesQueue) < queueSize {
+				// The use of the go routine is to avoid blocking
+				go func() {
+					nonce := <-requeueQueue
```

Comment:
can we just add this as a different case in the select/for loop?

Reply:
I don't want this to be blocking, that's why I added the condition. Also, we don't want to requeue every time so that we don't keep retry indefinitely the failing nonce, but instead do it once in an hour

Reply:
mind explaining how will it block? won't it only get triggered is something is pushed through the channel to the select statement? If it will block if noncesQueue is full, then imo we should either increase that buffer or empty the channel faster. I could be missing something ofc too

Reply:
perhaps the emptying and filling of the channel should not be in the same select statement?

Reply:
If nonces queue is full and we run:

```
noncesQueue <- nonce
```

it will block, and at the same time, it won't be able to process nonces because it will be stuck here, hence the use of the go routine.

Also, we're using the ticker so that the nonsense don't get processed every time they fail directly so that we give them an hour of a delay before they get re-processed.

> perhaps the emptying and filling of the channel should not be in the same select statement?

we could and it would be cleaner, but didn't want to add that extra complexity.

Reply:
why not just put the nonce in the queue when we attempt to retry instead of putting it in a channel, which eventually gets added to the nonce channel

or if we need the extra protection for some reason we can have a single goroutine that is connecting the two channels. 

this could just be me being too nit picky, but spinning up a single goroutine to connect two channels on a timer feels like there's room for improvement